### PR TITLE
Device Login Track-2 Migration 

### PR DESCRIPTION
**Pull request by pranavmalik-msft**
**Description**
This pull request aims to facilitate the migration process from the existing ADAL library to the MSAL framework, specifically for device login functionalities. 

**Changes**
- The existing flow for the ADAL framework has been removed regarding Device Login with the MSAL framework.
- All references to ADAL library has been removed -for token caching, refreshing and persistence.
- Adding new token persistence logic

### CODE REVIEW

File: common/oauthTokenManager.go [hunk 1]
```diff
@@ -245,41 +246,13 @@ func (uotm *UserOAuthTokenManager) UserLogin(tenantID, activeDirectoryEndpoint s
 		activeDirectoryEndpoint = DefaultActiveDirectoryEndpoint
 	}
 
-	// Init OAuth config
-	oauthConfig, err := adal.NewOAuthConfig(activeDirectoryEndpoint, tenantID)
+	dc, err := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{TenantID: tenantID})
```
Comment (vibhansa-msft) [inline common/oauthTokenManager.go:249]: In adal code we were able to provide the aad-endpoint as well, does track-2 not have option for that?

Comment (gapra-msft) [inline common/oauthTokenManager.go:249]: how does token persistence work across runs?

**Comment by nakulkar-msft** [comment]
This should be made a draft till we finalize the token persistence

**Comment by seanmcc-msft** [comment]
Open question - how will we test this? @adreed-msft any ideas?

File: common/oauthTokenManager.go (hunk 1 above)
Comment (pranavmalik-msft) [reply common/oauthTokenManager.go]: Added the token persistence logic

**Comment by adreed-msft** [comment]
> Open question - how will we test this? @adreed-msft any ideas?
> 
> Look into how Identity tests this.

The idea for testing was that we'd have semi-automated tests in the E2E suite. Stick the suite behind a build tag, something like "manualinteraction", run azcopy login at the suite setup, run logout at suite teardown.

I'd already given a little guidance in a 1:1 with Pranav.

**Comment by gapra-msft** [comment]
Can we close this PR in favor of the other one?

**Pull request closed without merging**